File: 8_August/Aug_27/gamer.js
const readline = require('readline');

class TextAdventureGame {
    constructor() {
        this.playerHealth = 100;
        this.playerGold = 0;
        this.enemies = ["Goblin", "Orc", "Troll", "Dragon"];
        this.locations = ["Forest", "Cave", "Castle", "Mountain"];
        this.rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });
    }

    async startGame() {
        console.log("Welcome to the Text Adventure Game!");
        console.log("You are a brave adventurer seeking fortune and glory.");
        
        while (this.playerHealth > 0) {
            await this.takeTurn();
        }
        
        console.log("Game Over! You have been defeated.");
        console.log(`You collected ${this.playerGold} gold during your adventure.`);
        this.rl.close();
    }

    async takeTurn() {
        const location = this.locations[Math.floor(Math.random() * this.locations.length)];
        console.log(`\nYou find yourself in a ${location}.`);
        
        const choice = await this.askQuestion("Do you want to [e]xplore or [r]est? ");
        
        if (choice.toLowerCase() === 'e') {
            await this.explore();
        } else if (choice.toLowerCase() === 'r') {
            this.rest();
        } else {
            console.log("Invalid choice. You stumble around and lose 5 health.");
            this.playerHealth -= 5;
        }
    }

    async explore() {
        if (Math.random() < 0.7) {  // 70% chance of encounter
            const enemy = this.enemies[Math.floor(Math.random() * this.enemies.length)];
            console.log(`You encounter a ${enemy}!`);
            
            const choice = await this.askQuestion("Do you want to [f]ight or [r]un? ");
            
            if (choice.toLowerCase() === 'f') {
                this.fight(enemy);
            } else if (choice.toLowerCase() === 'r') {
                this.run();
            } else {
                console.log("Invalid choice. You're caught off guard and lose 10 health.");
                this.playerHealth -= 10;
            }
        } else {
            const goldFound = Math.floor(Math.random() * 41) + 10;  // 10 to 50 gold
            this.playerGold += goldFound;
            console.log(`You found ${goldFound} gold!`);
        }
    }

    fight(enemy) {
        const enemyStrength = this.enemies.indexOf(enemy) * 10 + 10;
        const damageTaken = Math.floor(Math.random() * (enemyStrength + 1));
        this.playerHealth -= damageTaken;
        
        const goldWon = Math.floor(Math.random() * (enemyStrength + 1)) + enemyStrength;
        this.playerGold += goldWon;
        
        console.log(`You defeated the ${enemy}!`);
        console.log(`You took ${damageTaken} damage and won ${goldWon} gold.`);
        console.log(`Your health: ${this.playerHealth}`);
    }

    run() {
        const escapeChance = Math.random();
        if (escapeChance > 0.5) {
            console.log("You managed to escape!");
        } else {
            const damageTaken = Math.floor(Math.random() * 11) + 5;  // 5 to 15 damage
            this.playerHealth -= damageTaken;
            console.log(`You couldn't escape and took ${damageTaken} damage while running.`);
            console.log(`Your health: ${this.playerHealth}`);
        }
    }

    rest() {
        const healthGained = Math.floor(Math.random() * 11) + 5;  // 5 to 15 health
        this.playerHealth = Math.min(100, this.playerHealth + healthGained);
        console.log(`You rest and recover ${healthGained} health.`);
        console.log(`Your health: ${this.playerHealth}`);
    }

    askQuestion(question) {
        return new Promise((resolve) => {
            this.rl.question(question, (answer) => {
                resolve(answer);
            });
        });
    }
}

const game = new TextAdventureGame();
game.startGame();

File: 8_August/Aug_26/gamer.py
import random

class TextAdventureGame:
    def __init__(self):
        self.player_health = 100
        self.player_gold = 0
        self.enemies = ["Goblin", "Orc", "Troll", "Dragon"]
        self.locations = ["Forest", "Cave", "Castle", "Mountain"]

    def start_game(self):
        print("Welcome to the Text Adventure Game!")
        print("You are a brave adventurer seeking fortune and glory.")
        
        while self.player_health > 0:
            self.take_turn()
        
        print("Game Over! You have been defeated.")
        print(f"You collected {self.player_gold} gold during your adventure.")

    def take_turn(self):
        location = random.choice(self.locations)
        print(f"\nYou find yourself in a {location}.")
        
        choice = input("Do you want to [e]xplore or [r]est? ").lower()
        
        if choice == 'e':
            self.explore()
        elif choice == 'r':
            self.rest()
        else:
            print("Invalid choice. You stumble around and lose 5 health.")
            self.player_health -= 5

    def explore(self):
        if random.random() < 0.7:  # 70% chance of encounter
            enemy = random.choice(self.enemies)
            print(f"You encounter a {enemy}!")
            
            choice = input("Do you want to [f]ight or [r]un? ").lower()
            
            if choice == 'f':
                self.fight(enemy)
            elif choice == 'r':
                self.run()
            else:
                print("Invalid choice. You're caught off guard and lose 10 health.")
                self.player_health -= 10
        else:
            gold_found = random.randint(10, 50)
            self.player_gold += gold_found
            print(f"You found {gold_found} gold!")

    def fight(self, enemy):
        enemy_strength = self.enemies.index(enemy) * 10 + 10
        damage_taken = random.randint(0, enemy_strength)
        self.player_health -= damage_taken
        
        gold_won = random.randint(enemy_strength, enemy_strength * 2)
        self.player_gold += gold_won
        
        print(f"You defeated the {enemy}!")
        print(f"You took {damage_taken} damage and won {gold_won} gold.")
        print(f"Your health: {self.player_health}")

    def run(self):
        escape_chance = random.random()
        if escape_chance > 0.5:
            print("You managed to escape!")
        else:
            damage_taken = random.randint(5, 15)
            self.player_health -= damage_taken
            print(f"You couldn't escape and took {damage_taken} damage while running.")
            print(f"Your health: {self.player_health}")

    def rest(self):
        health_gained = random.randint(5, 15)
        self.player_health = min(100, self.player_health + health_gained)
        print(f"You rest and recover {health_gained} health.")
        print(f"Your health: {self.player_health}")

if __name__ == "__main__":
    game = TextAdventureGame()
    game.start_game()